pool-buffer.c

```diff
@@ -128,15 +128,10 @@ struct pool_buffer *get_next_buffer(struct wl_shm *shm,
 		return NULL;
 	}
 
-	if (buffer->width != width || buffer->height != height) {
-		destroy_buffer(buffer);
-	}
-
-	if (!buffer->buffer) {
-		if (!create_buffer(shm, buffer, width, height,
-					WL_SHM_FORMAT_ARGB8888)) {
+	destroy_buffer(buffer);
```

Comment:
The reasoning for this was due to me not figuring out how to properly reset the cairo surface to its' initial state before we start drawing a new one. With the old behaviour, we were basically overwriting the old contents, which is fine when you also redraw everything else, but not so much when you only want to redraw a single element. Since the indicator can be drawn half-transparent, the result of this is that you see the previously drawn indicator below the other. Simply deleting the buffer and creating a new one -every time- seemed to me like the by far simplest solution, though I guess a better approach would be resetting cairo surface somehow, or saving a copy of an empty surface to be copied into the existing one each time. I opted for what I felt would be simplest.

Reply:
Okay. Destroying the buffer each time is rather inefficient. Instead, you can clear the surface: https://github.com/emersion/mako/blob/master/render.c#L231